lib/quantizeAttributes.js

```diff
@@ -0,0 +1,348 @@
+                                        attribute.indexOf("TEXCOORD") >= 0 ||
+                                        attribute.indexOf("JOINT") >= 0 ||
+                                        attribute.indexOf("WEIGHT") >= 0 ||
+                                        attribute.indexOf("COLOR") >= 0
+                                    ) {
+                                        accessor = attributes[attribute];
+                                        accessorIds[accessor] = attribute;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Generate quantized attributes
+    for (var accessorId in accessorIds) {
+        if (accessorIds.hasOwnProperty(accessorId)) {
+            attribute = accessorIds[accessorId];
+            accessor = accessors[accessorId];
+            // as of right now, we only support 32-bit float to 16-bit int quantization
+            if (accessor.componentType != 5126) {
+                break;
+            }
+            accessor.extensions = {
+                "WEB3D_quantized_attributes": {}
+            };
+            attributes = accessor.extensions.WEB3D_quantized_attributes;
+            var accessorData = getAccessorData(gltf, accessorId);
+            var min = accessorData.min;
+            var max = accessorData.max;
+
+            var decodeMatrix = [];
+            var precision = Math.pow(2, 16) - 1;
+            for (j = 0; j < min.length + 1; j++) {
+                for (var k = 0; k < min.length + 1; k++) {
+                    if (j == min.length) {
+                        if (j == k) {
+                            decodeMatrix.push(1);
+                        }
+                        else {
+                            decodeMatrix.push(min[k]);
+                        }
+                    }
+                    else if (j == k) {
+                        decodeMatrix.push((max[j] - min[j]) / precision);
+                    }
+                    else {
+                        decodeMatrix.push(0);
+                    }
+                }
+            }
+            // decodedMin and decodedMax are required for POSITION attributes
+            if (attribute === "POSITION") {
+                attributes.decodedMin = min;
+                attributes.decodedMax = max;
+            }
+            attributes.decodeMatrix = decodeMatrix;
+
+
+            // Quantize the data
+            var data = accessorData.data;
+            var encoded = new Uint16Array(data.length);
+            var index;
+            for (j = 0; j < data.length; j++) {
+                index = j % min.length;
+                encoded[j] = (data[j] - min[index]) * precision / (max[index] - min[index]);
+            }
+
+            //TODO: Repack the buffers once, don't do it for each accessor
```

Comment:
There's nothing explicitly wrong with doing it the way it is, but it would probably be more efficient to reduce the amount of buffer modification. I'll try to get this fixed now.